moddio/moddio2
# Fix undo doesn't work as expect when fills empty tiles

## Summary
- fix debounce will cause undo create afterimage
- fix fill empty tiles will make undo behave not as expect
- fix undo putTiles will put 0 not -1 as empty tiles for phaser (phaser must use -1 as empty tile), it cause fill tool won't fill the whole map

## How to test
- I tested them all with 2 tabs, it seems works fine, and u can test in this order
1. just paint some tiles and undo them to see whether it would create afterimage or not
2. select walls layer and use taro.physics.enableDebug() /* use box2dwasm in client plz */
3. paint some tiles in walls layer and fill the empty tiles with the same tile, then undo it, the expect result is the painted tiles manually are still exist on the map
4.  use eraser to remove some tiles which painted in 3. but don't remove all of them, and fill the empty tiles with the same tile, the expect result is the whole map be filled with that tile
